cmd/srcd/cmd/sql.go

```diff
@@ -121,24 +126,39 @@ func repl() error {
 			return nil
 		case "":
 		default:
-			if err := runQuery(statement); err != nil {
+			if err := runQuery(client, statement); err != nil {
 				fmt.Println(err)
 			}
 		}
 	}
 }
 
-func runQuery(query string) error {
-	c, err := daemon.Client()
-	if err != nil {
-		return fmt.Errorf("could not get daemon client: %v", err)
+func ensureConnReady(client api.EngineClient) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	sleep := 1
+	for {
+		_, err := client.SQL(ctx, &api.SQLRequest{Query: "SELECT 1"})
```

Comment:
This is reusing the same `ctx` for each loop. If I'm not mistaken, when the context is cancelled after 30 seconds, any `client.SQL` call after that moment will always be started with an already cancelled context, and fail.